ocaml/typing/typedecl.ml

```diff
@@ -1610,6 +1607,17 @@ let update_decl_jkind env dpath decl =
       type_jkind
   in
 
+  (* Do not override the jkind if the manifest jkind if available. This is
+     necessary to support types like ['a or_null], which have a non-standard
+     jkind. [check_coherence] will update the jkind to match the manifest. *)
```

Comment:
I don't see how this comment corresponds to the implementation. It says not to do something if there is a manifest.... but the code does something if there is a manifest. Could you try rewording this?

Reply:
Edited.